src/main/java/seedu/address/model/insurance/UniqueLifeInsuranceMap.java

```diff
@@ -30,51 +31,51 @@
     private final ObservableList<ReadOnlyInsurance> internalList = FXCollections.observableArrayList();
 
     /**
-     * Returns true if the map contains an equivalent UUID as the given argument.
+     * Returns true if the map contains an equivalent key {@code UUID} as the given argument.
      */
     public boolean containsKey(UUID toCheck) {
         requireNonNull(toCheck);
         return internalMap.containsKey(toCheck);
     }
 
     /**
-     * Returns true if the map contains an equivalent insurance as the given argument.
+     * Returns true if the map contains an equivalent value {@code ReadOnlyInsurance} as the given argument.
      */
     public boolean containsValue(ReadOnlyInsurance toCheck) {
         requireNonNull(toCheck);
         return internalMap.containsValue(toCheck);
     }
 
     /**
+     * Returns true if an insurance inside the map contains an equivalent {@code contractName} as the given argument.
+     */
+    public boolean containsContractName(String toCheck) {
+        requireNonNull(toCheck);
+        return internalMap.values().stream().anyMatch(li ->
+            li.getContractName().equals(toCheck)
+        );
+    }
+
+    /**
      * Adds an life insurance to the map.
      *
      * @throws DuplicateInsuranceException if the life insurance to add is a duplicate of an
      * existing life insurance in the map.
      */
-    public void put(ReadOnlyInsurance toPut) throws DuplicateInsuranceException {
+    public void put(UUID key, ReadOnlyInsurance toPut)
+            throws DuplicateInsuranceException, DuplicateInsuranceContractNameException {
         requireNonNull(toPut);
         if (containsValue(toPut)) {
             throw new DuplicateInsuranceException();
         }
-        internalMap.put(toPut.getId(), new LifeInsurance(toPut));
-    }
-
-    /**
-     * Adds life insurance to map, private because this is only allowed when transferring data from another map
-     * @param id
-     * @param toPut
-     * @throws DuplicateInsuranceException
-     */
-    private void put(UUID id, ReadOnlyInsurance toPut) throws DuplicateInsuranceException {
-        requireNonNull(toPut);
-        if (containsValue(toPut)) {
-            throw new DuplicateInsuranceException();
+        if (containsContractName(toPut.getContractName())) {
+            throw new DuplicateInsuranceContractNameException();
         }
-        internalMap.put(id, new LifeInsurance(toPut));
+        internalMap.put(key, new LifeInsurance(toPut));
     }
```

Comment:
you deleted the overloaded method thats why there is a merge conflict and address book as well, pell from master, resolve and PR again

Reply:
Actually, I resolved the conflicts by deleting the overloaded method because I believe it's better to have two parameters: `key` and `value` for the `put` method.
However, I did ensure that no mismatching occurs by updating `Addressbook.java`. Please refer to the review below. Let me know of any concerns!